legacy/include/distconv/dnn_backend/convolution.hpp

```diff
@@ -648,47 +472,33 @@ class Convolution<BackendDNNLib, DataType>
-                                                 m_conv_fwd_d,
-                                                 m_fwd_algo,
-                                                 ws,
-                                                 m_ws_size_fwd,
-                                                 beta,
-                                                 output_proxy.desc(),
-                                                 output_proxy.ptr());
-               }
+                    m_be.convolution_forward(alpha,
+                                             m_input_d,
+                                             input_ptr,
+                                             m_filter_d,
+                                             filter.get_const_base_ptr(),
+                                             m_conv_fwd_d,
+                                             m_fwd_algo,
+                                             ws,
+                                             m_ws_size_fwd,
+                                             beta,
+                                             m_output_d,
+                                             output.get_base_ptr());
+                }
                 else
                 {
-                    auto input_proxy = dnn_lib::read_proxy(handle,m_input_d, input_ptr);
-                    auto output_proxy = dnn_lib::write_proxy(
-                        handle,
-                        m_output_d,
-                        output.get_base_ptr(),
-                        beta);
-                    backend::convolution_bwd_data(handle,
-                                                  alpha,
-                                                  m_filter_d,
-                                                  filter.get_const_base_ptr(),
-                                                  input_proxy.desc(),
-                                                  input_proxy.ptr(),
-                                                  m_conv_fwd_d,
-                                                  m_bwd_data_algo,
-                                                  ws,
-                                                  m_ws_size_fwd,
-                                                  beta,
-                                                  output_proxy.desc(),
-                                                  output_proxy.ptr());
+                    m_be.convolution_bwd_data(alpha,
+                                              m_filter_d,
+                                              filter.get_const_base_ptr(),
+                                              m_input_d,
+                                              input_ptr,
+                                              m_conv_fwd_d,
+                                              m_bwd_data_algo,
+                                              ws,
+                                              m_ws_size_fwd,
```

Comment:
I think within the context of this call, `m_ws_size_fwd` is actually correct since it's still computing a forward action. But yeah, this is outside the scope of this PR.